packages/custom_lint_core/lib/src/change_reporter.dart

```diff
@@ -76,9 +76,13 @@ abstract class ChangeBuilder {
   ///
   /// The builder passed to the [buildFileEdit] function has no special support
   /// for any particular kind of file.
+  ///
+  /// Use the [customPath] if the collection of edits should be written to another
+  /// file.
   void addGenericFileEdit(
-    void Function(analyzer_plugin.FileEditBuilder builder) buildFileEdit,
-  );
+    void Function(analyzer_plugin.FileEditBuilder builder) buildFileEdit, {
+    String? customPath,
```

Comment:
We'd need this on all `addXFileEdit`, such as addDartFileEdit

Reply:
Done. `customPath `parameter added to all `addXFileEdit` methods